### DISCUSSION TITLE
Bias in GLM projections for Minnesota

### DISCUSSION THREAD

**Discussion by hcorson-dosch-usgs** [DOI-USGS/lake-temperature-process-models#46] [General]
Starting a discussion to discuss potential bias in the GLM projections

--------------------------------

** Note: the uncalibrated GLM projections plotted here are from the initial [draft release](https://www.sciencebase.gov/catalog/item/6206d3c2d34ec05caca53071) -- Lindsay is working on generating the thermal metrics for the latest projections, which incorporate a fix for NA cells and a final resampling solution for hypsography **

--------------------------------

Relevant ScienceBase releases and items:
* [Midwest process-based water temperature predictions](https://www.sciencebase.gov/catalog/item/5db819bbe4b0b0c58b5a4c47) 
  * GLM version: 2
  * Thermal metrics file `'6_glm2pb0_annual_metrics.csv'`
* [MNTOHA](https://www.sciencebase.gov/catalog/item/5e774355e4b01d509270e2a1)
  * GLM version: 3.1
  * Thermal metrics file: `'7_pb0_annual_metrics.csv'`
* [Process-based projections](https://www.sciencebase.gov/catalog/item/6206d3c2d34ec05caca53071) for MN
  * GLM version: 3.2.0a8
  * Thermal metrics file: `'annual_thermal_metrics_glm2pb0_gcm.feather'`

**Comment by hcorson-dosch-usgs** [comment]
Gretchen and Holly have started digging into the initial MN projections we shared on the [draft release](https://www.sciencebase.gov/catalog/item/6206d3c2d34ec05caca53071) -- see Lake Temperature Modeling slack #general channel for discussion

When comparing the 1981-2000 mean of mean July surface temperatures between the GLM 3.2.0a8 projections (using GCM drivers) and the GLM2 predictions (using NLDAS drivers) , they noted that the projections are biased high by about 1.4 degrees across all GCMs:

<img src="https://user-images.githubusercontent.com/54007288/166719122-1ee5516f-d09e-4bd9-bdeb-a65cd2cc3f2f.png" width="600">

**Reply by hcorson-dosch-usgs** [reply]
The main difference between those 2 sets of predictions is of course the driver data sets. But since the GLM versions also differ, I wanted to do this same comparison between the MN projections and the MNTOHA predictions (GLM 3.1) as well as between the MNTOHA predictions and the midwest GLM 2 predictions.

MN projections (GLM 3.2.0a8) vs. midwest GLM 2 predictions (same plot as Gretchen and Holly made):
<img src="https://user-images.githubusercontent.com/54007288/166746098-f3043217-f9e2-4728-9f66-b7b5e6b9c3cf.png" width="500">

   * Mean across GCMs:
<img src="https://user-images.githubusercontent.com/54007288/166746154-0845ef3d-bf08-4f49-9eb3-f7b995261e9d.png" width="500">

MN projections (GLM 3.2.0a8) vs. MNTOHA GLM 3.1 predictions:
<img src="(https://user-images.githubusercontent.com/54007288/166746198-5c9bdb26-efe7-4212-8308-5cd8eb8b18d5.png" width="500">

  * Mean across GCMs:
<img src="https://user-images.githubusercontent.com/54007288/166746279-c17b5a69-f472-4110-a4f0-30281c327fed.png" width="500">

MNTOHA GLM 3.1 predictions vs. midwest GLM 2 predictions:
<img src="https://user-images.githubusercontent.com/54007288/166746347-7624b9de-f714-43fd-b1d7-e8d4069fb22a.png" width="500">

**Comment by hcorson-dosch-usgs** [comment]
### Seasonal comparisons

Ok trying to avoid a total deluge of plots, but this is pretty interesting. Looking now at mean surface temperatures for January, April, and October:
* January
<img src="https://user-images.githubusercontent.com/54007288/166746660-e6f4cc4d-5346-4ea4-90c1-28c4a757f202.png" width="500">

<img src="https://user-images.githubusercontent.com/54007288/166746729-03c44ebe-1476-488f-9c53-8227036abd82.png" width="500">

<img src="https://user-images.githubusercontent.com/54007288/166746768-acdf1fa2-781d-4994-9d8e-1eacc175198d.png" width="500">

* April
<img src="https://user-images.githubusercontent.com/54007288/166746898-3cc06d0e-7a69-4903-a0e9-3846d1991ab5.png" width="500">

<img src="https://user-images.githubusercontent.com/54007288/166746954-5351749d-546e-4c25-96c6-8b6a78d9d685.png" width="500">

<img src="https://user-images.githubusercontent.com/54007288/166746992-8b8456b3-4b12-413d-a1ae-08d9823fa16b.png" width="500">

* October
<img src="https://user-images.githubusercontent.com/54007288/166747407-f8a66463-08f0-498d-a961-f16852676fd9.png" width="500">

<img src="https://user-images.githubusercontent.com/54007288/166747444-d8e9f77b-ef87-4af3-bdcf-ed2cf9b942e2.png" width="500">

<img src="https://user-images.githubusercontent.com/54007288/166747487-78693a7b-8960-430c-9513-7cbe25288167.png" width="500">

**Comment by hcorson-dosch-usgs** [comment]
### Depth comparison

Mean bottom temperatures in July:
<img src="https://user-images.githubusercontent.com/54007288/166784826-d12594e2-6182-4613-a069-0f927160e9af.png" width="500">

<img src="https://user-images.githubusercontent.com/54007288/166785326-531d65f0-e34f-4755-b836-e14f5f8ee442.png" width="500">

<img src="https://user-images.githubusercontent.com/54007288/166786492-579d7fe6-535a-4876-a724-44fe725a931d.png" width="500">

**Comment by jordansread** [comment]
Re: bias, I'd suggest caution in digging too far into model vs model, especially with narrower windows (like months). The GCMs and their debiased outputs still have bias that is seasonal (NLDAS has issues too). And the models are imperfect so they can create seasonal bias too. One clear difference between GLM3 and GLM2 is the ice-cover, which will cause major differences in spring temperatures (I think you are seeing that for April). Also, the TOHA uses time-varying water clarity and the other simulations do not. The clarity has a seasonal component to it as well. 

What I would recommend is understanding bias of the actual observed temperatures vs predicted, which is found [here](https://www.sciencebase.gov/catalog/item/5e774324e4b01d509270e29f) for TOHA and does not exist for the process-based release (but does exist for Jared's release and I'd suggest we include it in all future ones too). 

We've also made bias plots for important dimensions of bias (time, season, temperature), e.g., here from Jared's paper
[bias_panel.pdf](https://github.com/USGS-R/lake-temperature-process-models/files/8625446/bias_panel.pdf)

But before digging too far into this it is important to understand what questions would be answered, as that informs the way to perform the data analysis. There will definitely be bias in the models and the inputs. There is also an acceptable amount of bias, which will vary depending on the metrics used and the end-user of the data (which is why we've started packaging modeled vs obs in the `eval` section as shared above, as this allows future users to look specifically at bias that may impact their particular use of the data, for example, looking at bias in warmer predictions because their conclusions would be impacted by that).

**Reply by hcorson-dosch-usgs** [reply]
Okay, thanks Jordan. That's helpful context, and makes sense that any model-model comparison here is limited in its utility. I figured that this was not a situation that warranted digging deeply, but wanted to take the time to plot a few variables similar to that Gretchen was exploring to get a handle on the relationships she was seeing. Is a formal evaluation step to compare predicted and observed temperatures (for the 1981-2000 contemporary predictions) something that you think is worth incorporating into this process modeling pipeline?

**Reply by jordansread** [reply]
I think model evaluation is a critical step. In past model runs, we've done those calculations as part of the data release repo, but it could live there or exist in a more generic form in the modeling repo(s). Past code in the data release repo to do this for NLDAS-driven GLMs was [here](https://github.com/USGS-R/mntoha-data-release/blob/main/6_evaluation.yml#L29). I'm not sure what I'd recommend in this case, since I don't know what the implementation trade-offs would be (for example, what other thing would be de-prioritized in order to add this?). 

But note that in the case of climate projections, even the contemporary period is based on climate (not weather), so the day to day match of values is not going to provide a useful estimate of accuracy, but could help estimate bias.
